index.js

```diff
@@ -30,26 +27,26 @@ module.exports = function(opts) {
     }
 
     if (!token && !opts.passthrough) {
-      this.throw(401, 'No authentication token found\n');
+      ctx.throw(401, 'No authentication token found\n');
     }
 
-    secret = (this.state && this.state.secret) ? this.state.secret : opts.secret;
+    secret = (ctx.state && ctx.state.secret) ? ctx.state.secret : opts.secret;
     if (!secret) {
-      this.throw(500, 'Invalid secret\n');
+      ctx.throw(500, 'Invalid secret\n');
     }
 
     try {
-      user = yield JWT.verify(token, secret, opts);
+      user = JWT.verify(token, secret, opts);
```

Comment:
Ah, sorry about that. I meant these parts in index.js (notice how `verify` gets thunkified:

```js
// Make verify function play nice with co/koa
var JWT = {decode: _JWT.decode, sign: _JWT.sign, verify: thunkify(_JWT.verify)};
```

and then later, at the bottom of the file:

```js
// Export JWT methods as a convenience
module.exports.sign   = _JWT.sign;
module.exports.verify = _JWT.verify;
module.exports.decode = _JWT.decode;
```

... which means that we export the thunkified `verify`, which is kind of messy.

This is discussed in #32.